packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -602,14 +603,14 @@ class _DwdsResidentWebRunner extends ResidentWebRunner {
           initializePlatform: debuggingOptions.initializePlatform,
           hostname: debuggingOptions.hostname,
           port: debuggingOptions.port,
-          skipDwds: device.device is WebServerDevice || !debuggingOptions.buildInfo.isDebug,
+          skipDwds: !deviceIsDebuggable || !debuggingOptions.buildInfo.isDebug,
           dartDefines: dartDefines,
         );
         // When connecting to a browser, update the message with a seemsSlow notification
         // to handle the case where we fail to connect.
         buildStatus.stop();
         statusActive = false;
-        if (debuggingOptions.browserLaunch && supportsServiceProtocol) {
+        if (supportsServiceProtocol) {
```

Comment:
This condition should be the same as skipDwds, since we can only exit from here if dwds is started.

Reply:
`supportsServiceProtocol` is defined as `isRunningDebug && deviceIsDebuggable`, so I think these are the same? Would it be better to change `skipDwds` to use `!supportsServiceProtocol`?

Reply:
Yeah - I think you could either make them the same or pull it into a single getter `_enableDwds` so they don't accidentally get out of sync

Reply:
Added a new getter, but also made its implementation use `debuggingEnabled` since it was the same condition and seems like they should match.